graphlearn/src/core/operator/sampler/condition_table.h

```diff
@@ -122,6 +122,20 @@ class ConditionTableFactory {
     }
   }
 
+  template<class T>
+  ConditionTable* LookupOrCreate(
```

Comment:
Multiple duplicate interfaces, we only need to keep 2 according to the previous interface？

Reply:
It is not duplicated. It is an overload for `io::Array<T>`. The reason for such an `io::Array` is that graphlearn internally use `std::vector<T>` for ids (`std::vector<int64_t>`) and datas (weights, `std::vector<float>`). However in vineyard those arrays are not `std::vector<T>` but a consecutive arrays on raw memory.

Thus we introduce the `io::Array` type for a view representation of the underlying raw memory. And we added overloads in `ConditionTable` and `AliasMethod`.

Reply:
It appears that the current code does not use these classes.